### PR TITLE
fix: Fetch MLS verification status UseCase [WPB-8610]

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

### Issues

Need a UseCase to force fetching the MLS verification status of specific conversation.

### Solutions

- extract from `MLSConversationsVerificationStatusesHandlerImpl` part that fetch the MLS verification status by `groupId` into separate internal UseCase
- create a new `FetchConversationMLSVerificationStatusUseCase` that is using a new internal useCase and will be called from platform
- update and add tests

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/FetchMLSVerificationStatusUseCase.kt [hunk 1]
```diff
@@ -65,62 +62,57 @@ data class VerificationStatusData(
...
-                                    newStatus =
-                                    ccGroupStatus
-                                )
-                            }
-                        }
-                    }.onSuccess {
-                        updateStatusAndNotifyUserIfNeeded(it)
+) : FetchMLSVerificationStatusUseCase {
+
+    private val logger = kaliumLogger.withTextTag("FetchMLSVerificationStatusUseCaseImpl")
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/FetchMLSVerificationStatusUseCase.kt:75]: question: do we use this logger anywhere?

**Review by mchenani — APPROVED** [review]
LGTM, added a question!

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/FetchMLSVerificationStatusUseCase.kt (hunk 1 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/FetchMLSVerificationStatusUseCase.kt:75]: yes, line 147, 167, 204 :)

Comment (alexandreferris) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/FetchMLSVerificationStatusUseCase.kt:75]: Doesn't look like its being used in this file anymore 🤔

**Review by saleniuk — APPROVED** [review]

**Pull request merged**